src/gpuarray_elemwise.c

```diff
@@ -31,6 +31,7 @@ struct _GpuElemwise {
 STATIC_ASSERT(GEN_CONVERT_F16 == GE_CONVERT_F16, same_flags_value_elem1);
 
 #define is_array(a) (ISCLR((a).flags, GE_SCALAR))
+#define is_rw_array(a) (ISSET((a).flags, GE_READ) && ISSET((a).flags, GE_WRITE))
```

Comment:
You should check for write only, we still don't want to broadcast output-only arguments.  Something like
`#define is_output(a) (ISSET((a).flags, GE_WRITE)`.